### ISSUE TITLE
[Feature] Stack order

### ISSUE THREAD

**Issue by rept0id** [colinl/node-red-guaranteed-delivery#2]
**_Hello!,_**

It would be nice if there were a way to select the order of accessing data from the buffer.

At present, the buffer is treated solely as a queue. However, some people might value the newest messages more than the oldest messages - while still wanting to transact all messages, for historical logging purposes. In this example, treating the data conditionally as a stack would be beneficial.

I'm looking forward to examining this idea further if you see its value.

**_Thank you,_**
_Rantouan Achmet_

**Comment by colinl** [comment]
> some people might value the newest messages more than the oldest

I have difficulty seeing that it makes much difference.  Once the network is back up one would expect the queue to be flushed in a few seconds anyway.

**Comment by colinl** [comment]
Also consider this situation.  Suppose the network is down and 50 messages become queued.  Then the network recovers and the first 25 are flushed.  Then suppose a new message comes in.  This would be added to the queue, but since messages are being flushed `newest first` it would be the next message sent, followed by the other 25.  The result would be that the messages were jumbled up.

**Comment by Gesinne** [comment]
I agree, more or less, with [rept0id](https://github.com/rept0id). And here’s another thought: Imagine this scenario: I have the “Max messages in queue” limit set based on the RPi’s RAM. Usually, they’re never offline long enough for the queue to fill up, but when it does happen, it would be good if, instead of stopping data storage, it replaced the old messages with the new ones. For future improvements. Thank you very much for your time.

**Comment by colinl** [comment]
> it would be good if, instead of stopping data storage, it replaced the old messages with the new ones. For future improvements.

That is covered by issue #1 
Unfortunately I don't have much time available at the moment so not much is happening on this node for the moment.

For the suggestion covered by this issue, did you see my second post on 17th May?

**Comment by Gesinne** [comment]
You are rigth. It is more related with issue https://github.com/colinl/node-red-guaranteed-delivery/issues/1. I will follow it as I said, as a suggestion for future improvements.

**Comment by colinl** [comment]
Closing this as I don't think it is a useful feature.
